packages/ui-core/src/components/paginated-table/components/body-cell/body-cell.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import { Cell as CellType } from 'react-table';
+
+import { Text } from '../../../../typography';
+import { Typography } from '../../../../types';
+import { CellTextAlignment, CellValue, FormattedValue } from '../../types';
+import { StyledCell, Container } from './body-cell.styles';
+
+type Props = {
+  cell: CellType;
+  typography: Typography;
+  idx: number;
+  onCellClick: (
+    e: React.MouseEvent<HTMLTableCellElement>,
+    columnName: string,
+    value: CellValue,
+    idx: number
+  ) => void;
+};
+
+const isFormattedValue = (el: any): el is FormattedValue =>
```

Comment:
As discussed - we could get rid of this after unifying table cell object interface.